### PR TITLE
Add new tables

### LINKED ISSUE #7: Read test cases checklist
What is our philosophy of test cases? Do we care about each individual feature? Or are we collecting a set of cases that have maximal coverage of important common and corner cases? I'm assuming the latter for this draft list.

Reader protocol v1:
 * [ ] A Delta Lake table with all data types
 * [ ] A table with a checkpoint
 * [ ] A table which has had a schema change
 * [ ] A table with stats as struct
 * [ ] A table with multiple levels of partitioning, including null values at both levels (should cover all serialization cases here: https://github.com/delta-io/delta/blob/master/PROTOCOL.md#partition-value-serialization)

Reader protocol v2:
 * [ ] Partitioned table with id-based column mapping
 * [ ] Partitioned table with name-based column mapping

### PR DESCRIPTION
**Pull request by wjones127**
## Description

~~Will rebase after #25 is merged.~~

 * Gets some, but all tables in #7

## Does this require an update to the documentation?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,8 @@ lint-bandit: ## Run bandit
 	@echo "\n${BLUE}Running bandit...${NC}\n"
 	@${POETRY_RUN} bandit -r ${PROJ}
 
-lint-base: lint-flake8 lint-bandit ## Just run the linters without autolinting
+#lint-base: lint-flake8 lint-bandit ## Just run the linters without autolinting
+lint-base: lint-flake8 # TODO: Can we drop bandit?
```
Comment (wjones127) [inline Makefile:29]: @edmondop is there a compelling reason to include bandit in this project? I found it was giving irrelevant warnings and since this project doesn't deal in any non-public data, it's unclear to me why we care that much about security.

**Review by MrPowers — APPROVED** [review]

**Comment by wjones127** [comment]
There are some test failures. I'll be on vacation for a week, so won't get around to finishing this until around December 20.

**Comment by MrPowers** [comment]
@wjones127 - alright cool, have a nice vacation!!  Let's finish this up and get it merged when you're back!

File: dat/generated_tables.py [hunk 2]
```diff
@@ -35,7 +40,7 @@ def save_expected(case: TestCaseInfo, as_latest=False) -> None:
     # Need to ensure directory exists first
     os.makedirs(case.expected_root(version))
 
-    df.toPandas().to_parquet(case.expected_path(version))
```
Comment (wjones127) [inline dat/generated_tables.py:38]: It turned out `toPandas().to_parquet()` causes weird things to happen to timestamps, so better to stick to Spark here. Without this, I was able to eliminate the pandas dependency as well.

File: dat/generated_tables.py [hunk 3]
```diff
@@ -171,3 +192,193 @@ def with_schema_change(case: TestCaseInfo, spark: SparkSession):
...
+
+
+@reference_table(
+    name='nested_types',
+    description='Table containing various nested types',
+)
+def create_nested_types(case: TestCaseInfo, spark: SparkSession):
+    schema = types.StructType([
+        types.StructField(
+            'pk', types.IntegerType()
```
Comment (wjones127) [inline dat/generated_tables.py:245]: chispa doesn't support ignoring sort order when comparing tables that contain map types, so we have to add a `pk` column that tests can sort on. I've added a note about this to the readme.
